modules/nf-core/merfin/hist/tests/main.nf.test

```diff
@@ -0,0 +1,297 @@
+                    [ id:'test_fragilis', single_end:false ], // meta map
+                    file(params.modules_testdata_base_path + '/genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz', checkIfExists: true)
+                    ])
+                    """
+                }
+            }
+
+            run("MERYL_HISTOGRAM") {
+                script "../../../meryl/histogram/main.nf"
+                config "./modules.config"
+                tag    "meryl/histogram"
+                process {
+                    """
+                    input[0] = MERYL_COUNT_READS.out.meryl_db
+                    """
+                }
+            }
+
+            run("GENOMESCOPE2") {
+                script "../../../genomescope2/main.nf"
+                config "./modules.config"
+                tag    "genomescope2"
+                process {
+                    """
+                    input[0] = MERYL_HISTOGRAM.out.hist
+                    """
+                }
+            }
+
+
+            
+            
+        }
+
+        when {
+            process {
+                """
+                Channel.from([
+                    [ id:'test_sars', single_end:false ], // meta map
+                    file(params.modules_testdata_base_path + '/genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
+                    ],[
+                    [ id:'test_fragilis', single_end:false ], // meta map
+                    file(params.modules_testdata_base_path + '/genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz', checkIfExists: true)
+                    ]).set{ input_zero_ch }
+
+                input_zero_ch.join( MERYL_COUNT_READS.out.meryl_db ).multiMap{ meta, asm, meryl_db ->
+                    asm_ch: [meta, asm]
+                    meryl_ch: [meta, meryl_db]
+                }.set{ multimap_inputs_ch }
+
+                // Due the fact that GENOMESCOPE2 module doesnt produce the optional output lookup_table.txt
+                // we had to do a hack to get this file's path
+                GENOMESCOPE2.out.summary.map{ meta, sum -> {
+                    def res = sum.getParent();
+                    [res/"lookup_table.txt"]
+                    }}.set{ lk_ch }
+                
+                input[0] = multimap_inputs_ch.asm_ch
+                input[1] = multimap_inputs_ch.meryl_ch
+                input[2] = lk_ch
+                input[3] = MERYL_COUNT_ASM.out.meryl_db.map{ meta, meryl_db -> meryl_db }
+                input[4] = 10
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out.get(0), process.out.get(2)).match() },
+                // The following asserts that the output stdout_log generated by the module is complete
+                { assert path(process.out.get(1).get(0)).readLines().last().contains("Bye!") }
+            )
+        }
+
+    }
```

Comment:
Nicely done, can you just please add a stub test?

Reply:
Finished. I also had to add `stub` section for the following modules: meryl/count, mery/histogram and genomescope2.